File: TwitterAI/src/modules/asset-matching/aliasDictionary.ts
import type { AssetType } from '@prisma/client';
import type { AliasSeed } from './assetMatching.types.js';

const SEEDS: AliasSeed[] = [
  // ── Mega-cap tech ──────────────────────────────────────────────
  { assetType: 'STOCK', ticker: 'TSLA', alias: 'tesla', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'NVDA', alias: 'nvidia', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'AAPL', alias: 'apple', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'MSFT', alias: 'microsoft', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'AMZN', alias: 'amazon', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'META', alias: 'meta', matchMethod: 'alias_dictionary', confidence: 0.65 },
  { assetType: 'STOCK', ticker: 'META', alias: 'facebook', matchMethod: 'alias_dictionary', confidence: 0.65 },
  { assetType: 'STOCK', ticker: 'GOOGL', alias: 'google', matchMethod: 'alias_dictionary', confidence: 0.65 },
  { assetType: 'STOCK', ticker: 'GOOGL', alias: 'alphabet', matchMethod: 'alias_dictionary', confidence: 0.65 },
  { assetType: 'STOCK', ticker: 'AMD', alias: 'amd', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'AMD', alias: 'advanced micro devices', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'NFLX', alias: 'netflix', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'ORCL', alias: 'oracle', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'CRM', alias: 'salesforce', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'ADBE', alias: 'adobe', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'INTC', alias: 'intel', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'QCOM', alias: 'qualcomm', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'AVGO', alias: 'broadcom', matchMethod: 'alias_dictionary', confidence: 0.75 },

  // ── Financials ────────────────────────────────────────────────
  { assetType: 'STOCK', ticker: 'JPM', alias: 'jpmorgan', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'JPM', alias: 'jp morgan', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'BAC', alias: 'bank of america', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'GS', alias: 'goldman sachs', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'GS', alias: 'goldman', matchMethod: 'alias_dictionary', confidence: 0.7 },
  { assetType: 'STOCK', ticker: 'MS', alias: 'morgan stanley', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'WFC', alias: 'wells fargo', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'C', alias: 'citigroup', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'C', alias: 'citi', matchMethod: 'alias_dictionary', confidence: 0.7 },
  { assetType: 'STOCK', ticker: 'V', alias: 'visa', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'MA', alias: 'mastercard', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'BRK.B', alias: 'berkshire', matchMethod: 'alias_dictionary', confidence: 0.7 },
  { assetType: 'STOCK', ticker: 'BRK.B', alias: 'berkshire hathaway', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'BLK', alias: 'blackrock', matchMethod: 'alias_dictionary', confidence: 0.8 },

  // ── Healthcare & pharma ────────────────────────────────────────
  { assetType: 'STOCK', ticker: 'UNH', alias: 'unitedhealth', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'JNJ', alias: 'johnson & johnson', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'JNJ', alias: 'johnson and johnson', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'PFE', alias: 'pfizer', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'MRK', alias: 'merck', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'ABBV', alias: 'abbvie', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'LLY', alias: 'eli lilly', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'LLY', alias: 'lilly', matchMethod: 'alias_dictionary', confidence: 0.65 },

  // ── Consumer & retail ──────────────────────────────────────────
  { assetType: 'STOCK', ticker: 'WMT', alias: 'walmart', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'TGT', alias: 'target', matchMethod: 'alias_dictionary', confidence: 0.7 },
  { assetType: 'STOCK', ticker: 'COST', alias: 'costco', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'HD', alias: 'home depot', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'MCD', alias: "mcdonald's", matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'SBUX', alias: 'starbucks', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'NKE', alias: 'nike', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'KO', alias: 'coca-cola', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'KO', alias: 'coca cola', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'PEP', alias: 'pepsi', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'PEP', alias: 'pepsico', matchMethod: 'alias_dictionary', confidence: 0.75 },

  // ── Energy ────────────────────────────────────────────────────
  { assetType: 'STOCK', ticker: 'XOM', alias: 'exxon', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'XOM', alias: 'exxonmobil', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'CVX', alias: 'chevron', matchMethod: 'alias_dictionary', confidence: 0.8 },

  // ── Industrials / defence / aerospace ─────────────────────────
  { assetType: 'STOCK', ticker: 'BA', alias: 'boeing', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'CAT', alias: 'caterpillar', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'DE', alias: 'deere', matchMethod: 'alias_dictionary', confidence: 0.65 },
  { assetType: 'STOCK', ticker: 'LMT', alias: 'lockheed martin', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'RTX', alias: 'raytheon', matchMethod: 'alias_dictionary', confidence: 0.75 },

  // ── Media / entertainment ─────────────────────────────────────
  { assetType: 'STOCK', ticker: 'DIS', alias: 'disney', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'PARA', alias: 'paramount', matchMethod: 'alias_dictionary', confidence: 0.7 },
  { assetType: 'STOCK', ticker: 'WBD', alias: 'warner bros', matchMethod: 'alias_dictionary', confidence: 0.7 },

  // ── Other notable names ───────────────────────────────────────
  { assetType: 'STOCK', ticker: 'COIN', alias: 'coinbase', matchMethod: 'alias_dictionary', confidence: 0.85 },
  { assetType: 'STOCK', ticker: 'MSTR', alias: 'microstrategy', matchMethod: 'alias_dictionary', confidence: 0.85 },
  { assetType: 'STOCK', ticker: 'PLTR', alias: 'palantir', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'RBLX', alias: 'roblox', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'SNAP', alias: 'snapchat', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'UBER', alias: 'uber', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'LYFT', alias: 'lyft', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'ABNB', alias: 'airbnb', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'HOOD', alias: 'robinhood', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'RIVN', alias: 'rivian', matchMethod: 'alias_dictionary', confidence: 0.8 },
  { assetType: 'STOCK', ticker: 'LCID', alias: 'lucid', matchMethod: 'alias_dictionary', confidence: 0.75 },
  { assetType: 'STOCK', ticker: 'GME', alias: 'gamestop', matchMethod: 'alias_dictionary', confidence: 0.85 },
  { assetType: 'STOCK', ticker: 'AMC', alias: 'amc', matchMethod: 'alias_dictionary', confidence: 0.7 },

  // ── ETFs / benchmarks ─────────────────────────────────────────
  { assetType: 'ETF', ticker: 'SPY', alias: 's&p 500', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'SPY', alias: 'sp 500', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'SPY', alias: 's&p500', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'SPY', alias: 'sp500', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'QQQ', alias: 'nasdaq', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'QQQ', alias: 'nasdaq 100', matchMethod: 'keyword_rule', confidence: 0.75 },
  { assetType: 'ETF', ticker: 'QQQ', alias: 'nasdaq100', matchMethod: 'keyword_rule', confidence: 0.75 },
  { assetType: 'ETF', ticker: 'IWM', alias: 'russell 2000', matchMethod: 'keyword_rule', confidence: 0.75 },
  { assetType: 'ETF', ticker: 'GLD', alias: 'gold etf', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'GLD', alias: 'gold price', matchMethod: 'keyword_rule', confidence: 0.6 },
  { assetType: 'ETF', ticker: 'SLV', alias: 'silver etf', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'TLT', alias: '20 year treasury', matchMethod: 'keyword_rule', confidence: 0.7 },
  { assetType: 'ETF', ticker: 'TLT', alias: 'long bond', matchMethod: 'keyword_rule', confidence: 0.5 },
  { assetType: 'ETF', ticker: 'XLF', alias: 'financial sector', matchMethod: 'keyword_rule', confidence: 0.55 },
  { assetType: 'ETF', ticker: 'XLE', alias: 'energy sector', matchMethod: 'keyword_rule', confidence: 0.55 },
  { assetType: 'ETF', ticker: 'XLK', alias: 'tech sector', matchMethod: 'keyword_rule', confidence: 0.5 },
  { assetType: 'ETF', ticker: 'ARKK', alias: 'ark invest', matchMethod: 'keyword_rule', confidence: 0.75 },

  // Macro-ish terms (treated as benchmarks / proxies)
  { assetType: 'ETF', ticker: 'SPY', alias: 'stocks', matchMethod: 'keyword_rule', confidence: 0.4 },
  { assetType: 'ETF', ticker: 'QQQ', alias: 'tech stocks', matchMethod: 'keyword_rule', confidence: 0.45 },
  { assetType: 'ETF', ticker: 'SPY', alias: 'stock market', matchMethod: 'keyword_rule', confidence: 0.45 },
  { assetType: 'ETF', ticker: 'SPY', alias: 'equities', matchMethod: 'keyword_rule', confidence: 0.4 },
  { assetType: 'ETF', ticker: 'GLD', alias: 'gold', matchMethod: 'keyword_rule', confidence: 0.5 },

  // ── Crypto ────────────────────────────────────────────────────
  { assetType: 'CRYPTO', ticker: 'BTC', alias: 'bitcoin', matchMethod: 'crypto_alias', confidence: 0.85 },
  { assetType: 'CRYPTO', ticker: 'BTC', alias: 'btc', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'ETH', alias: 'ethereum', matchMethod: 'crypto_alias', confidence: 0.85 },
  { assetType: 'CRYPTO', ticker: 'ETH', alias: 'eth', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'SOL', alias: 'solana', matchMethod: 'crypto_alias', confidence: 0.85 },
  { assetType: 'CRYPTO', ticker: 'SOL', alias: 'sol', matchMethod: 'crypto_alias', confidence: 0.75 },
  { assetType: 'CRYPTO', ticker: 'XRP', alias: 'ripple', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'XRP', alias: 'xrp', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'ADA', alias: 'cardano', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'DOGE', alias: 'dogecoin', matchMethod: 'crypto_alias', confidence: 0.85 },
  { assetType: 'CRYPTO', ticker: 'DOGE', alias: 'doge', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'AVAX', alias: 'avalanche', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'LINK', alias: 'chainlink', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'DOT', alias: 'polkadot', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'MATIC', alias: 'polygon', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'LTC', alias: 'litecoin', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'UNI', alias: 'uniswap', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'SHIB', alias: 'shiba inu', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'SHIB', alias: 'shib', matchMethod: 'crypto_alias', confidence: 0.75 },
  { assetType: 'CRYPTO', ticker: 'BNB', alias: 'binance coin', matchMethod: 'crypto_alias', confidence: 0.8 },
  { assetType: 'CRYPTO', ticker: 'BNB', alias: 'bnb', matchMethod: 'crypto_alias', confidence: 0.75 },
  { assetType: 'CRYPTO', ticker: 'SUI', alias: 'sui', matchMethod: 'crypto_alias', confidence: 0.75 },
  { assetType: 'CRYPTO', ticker: 'APT', alias: 'aptos', matchMethod: 'crypto_alias', confidence: 0.75 },
  { assetType: 'CRYPTO', ticker: 'PEPE', alias: 'pepe', matchMethod: 'crypto_alias', confidence: 0.7 },
];

export function seededAliases(): AliasSeed[] {
  // Defensive copy to avoid accidental mutation by callers.
  return SEEDS.map((x) => ({ ...x }));
}

export const KNOWN_TICKERS: string[] = Array.from(
  new Set(
    SEEDS.filter((s) => s.assetType === ('STOCK' satisfies AssetType) || s.assetType === 'ETF')
      .map((s) => s.ticker.toUpperCase())
      .concat([
        'SPY', 'QQQ', 'IWM', 'GLD', 'SLV', 'TLT', 'XLF', 'XLE', 'XLK', 'ARKK',
        'TSLA', 'NVDA', 'AAPL', 'MSFT', 'AMZN', 'META', 'GOOGL', 'GOOG', 'AMD',
        'NFLX', 'ORCL', 'CRM', 'ADBE', 'INTC', 'QCOM', 'AVGO',
        'JPM', 'BAC', 'GS', 'MS', 'WFC', 'C', 'V', 'MA', 'BLK',
        'UNH', 'JNJ', 'PFE', 'MRK', 'ABBV', 'LLY',
        'WMT', 'TGT', 'COST', 'HD', 'MCD', 'SBUX', 'NKE', 'KO', 'PEP',
        'XOM', 'CVX', 'BA', 'CAT', 'DE', 'LMT', 'RTX',
        'DIS', 'PARA', 'WBD',
        'COIN', 'MSTR', 'PLTR', 'RBLX', 'SNAP', 'UBER', 'LYFT', 'ABNB',
        'HOOD', 'RIVN', 'LCID', 'GME', 'AMC',
      ])
  )
);

export const KNOWN_CRYPTO_TICKERS: string[] = Array.from(
  new Set(
    SEEDS.filter((s) => s.assetType === 'CRYPTO')
      .map((s) => s.ticker.toUpperCase())
      .concat(['BTC', 'ETH', 'SOL', 'XRP', 'ADA', 'DOGE', 'AVAX', 'LINK', 'DOT', 'MATIC', 'LTC', 'UNI', 'SHIB', 'BNB', 'SUI', 'APT', 'PEPE'])
  )
);

File: AI/TwitterAI/pipeline/alias_dictionary.py
"""
Static alias seed dictionary — company names, ETF keywords, crypto aliases.
Extended at runtime from the asset_aliases DB table.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

MatchMethod = Literal["alias_dictionary", "keyword_rule", "crypto_alias", "cashtag", "direct_ticker", "db"]
AssetTypeLit = Literal["STOCK", "ETF", "CRYPTO", "INDEX", "FX", "COMMODITY", "UNKNOWN"]


@dataclass
class AliasSeed:
    asset_type: AssetTypeLit
    ticker: str
    alias: str
    match_method: MatchMethod
    confidence: float


_SEEDS: list[AliasSeed] = [
    # ── Mega-cap tech ──────────────────────────────────────────────
    AliasSeed("STOCK", "TSLA", "tesla", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "NVDA", "nvidia", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "AAPL", "apple", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "MSFT", "microsoft", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "AMZN", "amazon", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "META", "meta", "alias_dictionary", 0.65),
    AliasSeed("STOCK", "META", "facebook", "alias_dictionary", 0.65),
    AliasSeed("STOCK", "GOOGL", "google", "alias_dictionary", 0.65),
    AliasSeed("STOCK", "GOOGL", "alphabet", "alias_dictionary", 0.65),
    AliasSeed("STOCK", "AMD", "amd", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "AMD", "advanced micro devices", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "NFLX", "netflix", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "ORCL", "oracle", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "CRM", "salesforce", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "ADBE", "adobe", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "INTC", "intel", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "QCOM", "qualcomm", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "AVGO", "broadcom", "alias_dictionary", 0.75),
    # ── Financials ────────────────────────────────────────────────
    AliasSeed("STOCK", "JPM", "jpmorgan", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "JPM", "jp morgan", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "BAC", "bank of america", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "GS", "goldman sachs", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "GS", "goldman", "alias_dictionary", 0.7),
    AliasSeed("STOCK", "MS", "morgan stanley", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "WFC", "wells fargo", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "C", "citigroup", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "C", "citi", "alias_dictionary", 0.7),
    AliasSeed("STOCK", "V", "visa", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "MA", "mastercard", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "BRK.B", "berkshire", "alias_dictionary", 0.7),
    AliasSeed("STOCK", "BRK.B", "berkshire hathaway", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "BLK", "blackrock", "alias_dictionary", 0.8),
    # ── Healthcare & pharma ────────────────────────────────────────
    AliasSeed("STOCK", "UNH", "unitedhealth", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "JNJ", "johnson & johnson", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "JNJ", "johnson and johnson", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "PFE", "pfizer", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "MRK", "merck", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "ABBV", "abbvie", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "LLY", "eli lilly", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "LLY", "lilly", "alias_dictionary", 0.65),
    # ── Consumer & retail ──────────────────────────────────────────
    AliasSeed("STOCK", "WMT", "walmart", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "TGT", "target", "alias_dictionary", 0.7),
    AliasSeed("STOCK", "COST", "costco", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "HD", "home depot", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "MCD", "mcdonald's", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "SBUX", "starbucks", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "NKE", "nike", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "KO", "coca-cola", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "KO", "coca cola", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "PEP", "pepsi", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "PEP", "pepsico", "alias_dictionary", 0.75),
    # ── Energy ────────────────────────────────────────────────────
    AliasSeed("STOCK", "XOM", "exxon", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "XOM", "exxonmobil", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "CVX", "chevron", "alias_dictionary", 0.8),
    # ── Industrials / defence / aerospace ─────────────────────────
    AliasSeed("STOCK", "BA", "boeing", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "CAT", "caterpillar", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "LMT", "lockheed martin", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "RTX", "raytheon", "alias_dictionary", 0.75),
    # ── Media / entertainment ─────────────────────────────────────
    AliasSeed("STOCK", "DIS", "disney", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "PARA", "paramount", "alias_dictionary", 0.7),
    AliasSeed("STOCK", "WBD", "warner bros", "alias_dictionary", 0.7),
    # ── Other notable names ───────────────────────────────────────
    AliasSeed("STOCK", "COIN", "coinbase", "alias_dictionary", 0.85),
    AliasSeed("STOCK", "MSTR", "microstrategy", "alias_dictionary", 0.85),
    AliasSeed("STOCK", "PLTR", "palantir", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "RBLX", "roblox", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "SNAP", "snapchat", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "UBER", "uber", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "LYFT", "lyft", "alias_dictionary", 0.75),
    AliasSeed("STOCK", "ABNB", "airbnb", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "HOOD", "robinhood", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "RIVN", "rivian", "alias_dictionary", 0.8),
    AliasSeed("STOCK", "GME", "gamestop", "alias_dictionary", 0.85),
    AliasSeed("STOCK", "AMC", "amc", "alias_dictionary", 0.7),
    # ── ETFs / benchmarks ─────────────────────────────────────────
    AliasSeed("ETF", "SPY", "s&p 500", "keyword_rule", 0.7),
    AliasSeed("ETF", "SPY", "sp 500", "keyword_rule", 0.7),
    AliasSeed("ETF", "SPY", "s&p500", "keyword_rule", 0.7),
    AliasSeed("ETF", "SPY", "sp500", "keyword_rule", 0.7),
    AliasSeed("ETF", "QQQ", "nasdaq", "keyword_rule", 0.7),
    AliasSeed("ETF", "QQQ", "nasdaq 100", "keyword_rule", 0.75),
    AliasSeed("ETF", "QQQ", "nasdaq100", "keyword_rule", 0.75),
    AliasSeed("ETF", "IWM", "russell 2000", "keyword_rule", 0.75),
    AliasSeed("ETF", "GLD", "gold etf", "keyword_rule", 0.7),
    AliasSeed("ETF", "GLD", "gold price", "keyword_rule", 0.6),
    AliasSeed("ETF", "SLV", "silver etf", "keyword_rule", 0.7),
    AliasSeed("ETF", "TLT", "20 year treasury", "keyword_rule", 0.7),
    AliasSeed("ETF", "TLT", "long bond", "keyword_rule", 0.5),
    AliasSeed("ETF", "XLF", "financial sector", "keyword_rule", 0.55),
    AliasSeed("ETF", "XLE", "energy sector", "keyword_rule", 0.55),
    AliasSeed("ETF", "XLK", "tech sector", "keyword_rule", 0.5),
    AliasSeed("ETF", "ARKK", "ark invest", "keyword_rule", 0.75),
    AliasSeed("ETF", "SPY", "stocks", "keyword_rule", 0.4),
    AliasSeed("ETF", "QQQ", "tech stocks", "keyword_rule", 0.45),
    AliasSeed("ETF", "SPY", "stock market", "keyword_rule", 0.45),
    AliasSeed("ETF", "SPY", "equities", "keyword_rule", 0.4),
    AliasSeed("ETF", "GLD", "gold", "keyword_rule", 0.5),
    # ── Crypto ────────────────────────────────────────────────────
    AliasSeed("CRYPTO", "BTC", "bitcoin", "crypto_alias", 0.85),
    AliasSeed("CRYPTO", "BTC", "btc", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "ETH", "ethereum", "crypto_alias", 0.85),
    AliasSeed("CRYPTO", "ETH", "eth", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "SOL", "solana", "crypto_alias", 0.85),
    AliasSeed("CRYPTO", "SOL", "sol", "crypto_alias", 0.75),
    AliasSeed("CRYPTO", "XRP", "ripple", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "XRP", "xrp", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "ADA", "cardano", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "DOGE", "dogecoin", "crypto_alias", 0.85),
    AliasSeed("CRYPTO", "DOGE", "doge", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "AVAX", "avalanche", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "LINK", "chainlink", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "DOT", "polkadot", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "MATIC", "polygon", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "LTC", "litecoin", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "UNI", "uniswap", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "SHIB", "shiba inu", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "SHIB", "shib", "crypto_alias", 0.75),
    AliasSeed("CRYPTO", "BNB", "binance coin", "crypto_alias", 0.8),
    AliasSeed("CRYPTO", "BNB", "bnb", "crypto_alias", 0.75),
    AliasSeed("CRYPTO", "SUI", "sui", "crypto_alias", 0.75),
    AliasSeed("CRYPTO", "APT", "aptos", "crypto_alias", 0.75),
    AliasSeed("CRYPTO", "PEPE", "pepe", "crypto_alias", 0.7),
]

KNOWN_TICKERS: set[str] = set()
KNOWN_CRYPTO_TICKERS: set[str] = set()

for _s in _SEEDS:
    if _s.asset_type in ("STOCK", "ETF"):
        KNOWN_TICKERS.add(_s.ticker.upper())
    elif _s.asset_type == "CRYPTO":
        KNOWN_CRYPTO_TICKERS.add(_s.ticker.upper())

# Extra well-known tickers not already covered
KNOWN_TICKERS.update([
    "SPY", "QQQ", "IWM", "GLD", "SLV", "TLT", "XLF", "XLE", "XLK", "ARKK",
    "TSLA", "NVDA", "AAPL", "MSFT", "AMZN", "META", "GOOGL", "GOOG", "AMD",
    "NFLX", "ORCL", "CRM", "ADBE", "INTC", "QCOM", "AVGO",
    "JPM", "BAC", "GS", "MS", "WFC", "C", "V", "MA", "BLK",
    "UNH", "JNJ", "PFE", "MRK", "ABBV", "LLY",
    "WMT", "TGT", "COST", "HD", "MCD", "SBUX", "NKE", "KO", "PEP",
    "XOM", "CVX", "BA", "CAT", "DE", "LMT", "RTX",
    "DIS", "PARA", "WBD",
    "COIN", "MSTR", "PLTR", "RBLX", "SNAP", "UBER", "LYFT", "ABNB",
    "HOOD", "RIVN", "LCID", "GME", "AMC",
])
KNOWN_CRYPTO_TICKERS.update([
    "BTC", "ETH", "SOL", "XRP", "ADA", "DOGE", "AVAX", "LINK",
    "DOT", "MATIC", "LTC", "UNI", "SHIB", "BNB", "SUI", "APT", "PEPE",
])


def get_seeds() -> list[AliasSeed]:
    return list(_SEEDS)
